dashboard/lib/service/appengine_cocoon.dart

```diff
@@ -146,23 +146,33 @@ class AppEngineCocoonService implements CocoonService {
   }
 
   @override
-  Future<bool> rerunTask(Task task, String idToken) async {
-    assert(idToken != null);
+  Future<CocoonResponse<bool>> rerunTask(Task task, String idToken) async {
+    if (idToken == null || idToken.isEmpty) {
+      return const CocoonResponse<bool>.error('Need to sign in to perform rerun operation');
+    }
 
     final QualifiedTask qualifiedTask = QualifiedTask.fromTask(task);
     assert(qualifiedTask.isLuci);
 
-    /// This endpoint only returns a status code.
+    //this endpoint returns CocoonResponse<bool> which reflects whether the api
+    //call returns success, or returns bad status code.
     final Uri postResetTaskUrl = apiEndpoint('/api/reset-prod-task');
+
     final http.Response response = await _client.post(postResetTaskUrl,
         headers: <String, String>{
           'X-Flutter-IdToken': idToken,
         },
         body: jsonEncode(<String, String>{
           'Key': task.key.child.name,
         }));
-
-    return response.statusCode == HttpStatus.ok;
+    final bool success = response.statusCode == HttpStatus.ok;
+    return success
+        ? const CocoonResponse<bool>.data(true)
+        : CocoonResponse<bool>.error('bad status code is ' +
```

Comment:
ditto about useful error messages

For now, let's use the error message being sent from the backend

https://cs.opensource.google/flutter/cocoon/+/main:app_dart/lib/src/request_handlers/reset_prod_task.dart

Reply:
I might be wrong, my understanding is that the error message sent from this handler is automatically encoded in response.body, so that we can retrieve from response.body? I am not 100 percent clear on the underlying mechanisms of how the request_handlers are registered and managed

Reply:
This sgtm! This is to acknowledge that we'll need a follow up PR in the backend to make the error messages there more useful (this PR will only surface the messages, not make them useful)

Reply:
Oh Casey it's so kind of you to take time to review while still on jury duty. Feel free to reply when you are more convenient. And thanks a lot for the guidance!